File: 167.TwoSumII/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        if (numbers.size()<1)
        	return numbers;

        int i=0;
        int j=numbers.size()-1;
        while (i<j) {
        	if (target < (numbers[i] + numbers[j])){
        		j--;
        	}
        	else if (target > (numbers[i]+numbers[j])){
        		i++;
        	}
        	else{
        		return {i+1,j+1};
        	}
        }
        return {i++, j++};
    }
};

File: 167.TwoSumII/TwoSum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
 		int[] res = new int[2];
 		if (numbers == null || numbers.length < 2) {
 			return numbers;
 		}       
    	Map<Integer, Integer> remainSet= new HashMap<Integer, Integer>();
    	for (int i = 0; i<numbers.length; i++) {
    		if (remainSet.containsKey(numbers[i])) {
    			res[0] = remainSet.get(numbers[i]) + 1;
    			res[1] = i + 1;
    			return res;
    		}
    		else {
    			remainSet.put(target - numbers[i],i);
    		}
    	}
    	return res;
    }
}
